straxen/plugins/acqmon_processing.py

```diff
@@ -0,0 +1,132 @@
+
+@export
+@strax.takes_config(strax.Option('hit_min_amplitude_aqmon', default = 50, track = False,
+                                 help = 'Minimum hit threshold in ADC*counts above baseline'),
+                    strax.Option('baseline_samples_aqmon', default = 10, track = False,
+                                 help = 'Number of samples to use at the start of the pulse to determine the baseline'))
+
+class AqmonHits(strax.Plugin):
+    """ Find hits in acquisition monitor data. These hits could be 
+        then used by other plugins for deadtime calculations, 
+        GPS SYNC analysis, etc.
+    """
+    __version__ ='0.0.3'
+        
+    depends_on = ('raw_records_aqmon') 
+    provides  = ('aqmon_hits')
+    data_kind = ('aqmon_hits')
+    
+    dtype = strax.hit_dtype
+
+    save_when = strax.SaveWhen.TARGET
+
+    def compute(self, raw_records_aqmon):
+        rec = strax.raw_to_records(raw_records_aqmon)
+        strax.zero_out_of_bounds(rec)
+        strax.baseline(rec, baseline_samples = self.config['baseline_samples_aqmon'], flip = True)
+        aqmon_hits = strax.find_hits(rec, min_amplitude = self.config['hit_min_amplitude_aqmon'])
+    
+        return aqmon_hits 
+
+    
+
+#### Veto hardware ####:
+# V1495 busy veto module: 
+# Generates a 25 ns NIM pulse whenever a veto begins and a 25 ns NIM signal when it ends. 
+# A new start signal can occur only after the previous busy instance ended.
+# 1ms (1e6 ns) - minimum busy veto length, or until the board clears its memory
+    
+# DDC10 High Energy Veto:
+# 10ms (1e7 ns) - fixed HE veto length in XENON1T DDC10, 
+# in XENONnT it will be calibrated based on the length of large S2 SE tails
+# The start/stop signals for the HEV are generated by the V1495 board
+  
+@export
+@strax.takes_config(
+    strax.Option('max_veto_gap', default = int(5e8),
+                 help = 'Maximum separation between veto stop and start pulses [ns]'),
+    strax.Option('channel_map', track = False, type = immutabledict, 
+                 help = 'immutabledict mapping subdetector to (min, max)'
+                      'channel number.'))
+
+class VetoIntervals(strax.OverlapWindowPlugin):
+    """ Find pairs of veto start and veto stop signals and the veto duration between them
+    busy_*  <= V1495 busy veto for tpc channels
+    he_*    <= V1495 busy veto for high energy tpc channels
+    hev_*   <= DDC10 hardware high energy veto
+    """
+        
+    __version__ ='0.0.23'
+    depends_on = ('aqmon_hits') 
+    provides  = ('veto_intervals')
+    data_kind = ('veto_intervals')
+
+    def infer_dtype(self):
+        dtype = [(('veto start time since unix epoch [ns]', 'time'), np.int64),
+                 (('veto end time since unix epoch [ns]', 'endtime'), np.int64),
+                 (('veto interval [ns]', 'veto_interval'), np.int64),
+                 (('veto signal type', 'veto_type'), np.str('U10'))]
+        return dtype
+    
+    def setup(self):
+        aq_ch_start = 800
```

Comment:
This was suggested here by Joran: https://github.com/XENONnT/straxen/pull/207#discussion_r486989546
I highly doubt that we are going to change the channel map or hardware cabling at this point, and if we do any such change will still mean that we need to change all the mapping. 

Nonetheless, my suggestion would be to split the aqmon channels in the straxen context, https://github.com/AlexElykov/straxen/blob/56abca546b2e6f31ff3a49b04513cd6ba3fe00bb/straxen/contexts.py#L31
to something like this: 

soft_aqmon = (790,799)    
hard_aqmon = (800, 807)

Then I can use the hard_aqmon and it would be easy to track the changes.

Reply:
Okay I see, but maybe we can then find a better place for this hard coded number? I think so far we put all hard coded value which we should keep track of in the beaning of  common.py. Would this be okay with you?

Reply:
Daniel I'm not sure if you do see Alex's point. This plugin needs the hardware AM channels, but the global channel map doesn't distinguish between the hardware and software channels, it just lumps everything together.

Reply:
No No thanks, this I understood from Alex comment. And I agree on his point, that we should not split up the aqmon channels on the raw_records level. I just would like to have a good bookkeeping of all really important hard coded numbers. I think this is the case here since in principle I can imagine that also some other plugin may distinguish between the "soft" and "hard" aqmon channels. And hence having both plugins, functions or whatever referring to the same parameter in common.py will give us less headaches in case we change something (even though this might be unlikely, but you never know).

Reply:
As far as I know there's nothing past DAQreader that uses the software channels, and this is where all the software-based deadtime signals show up from both strax and redax. Currently these are ignored. I agree that we don't want separate files for the SW and HW components, but it'll be really handy to have this distinction hard-coded somewhere sensible. If this sensible location is common.py or in the context, I don't know.

Reply:
Alright, then I'll just add it to common.py as a variable. Any requests for its name. Something like "n_hard_aqmon_start = 800" would be fine with you @WenzDaniel ?